bigtable/src/main/scala/magnolify/bigtable/BigtableType.scala

```diff
@@ -14,88 +14,130 @@
-               timestamp: Long = 0L,
-               columnQualifier: String = null): Iterable[Mutation]
+sealed trait BigtableType[T] extends Converter[T, Row, Seq[Mutation]] {
+  protected val columnFamily: String
+  protected val timestampMicros: Long
 
+  def apply(v: Row): T = from(v)
+  def apply(v: T): Seq[Mutation] = to(v)
 }
 
 object BigtableType {
-  val DEFAULT_COLUMN_QUALIFIER_NAME = "q"
   val DEFAULT_COLUMN_FAMILY_NAME = "f"
+  val DEFAULT_COLUMN_QUALIFIER_NAME = "q"
 
-  implicit def apply[T](implicit f: BigtableField[T]): BigtableType[T] = new BigtableType[T] {
+  implicit def apply[T](implicit f: BigtableField[T]): BigtableType[T] =
+    BigtableType(BigtableType.DEFAULT_COLUMN_FAMILY_NAME, 0L)
 
-    def from(v: Row, columnFamily: String = BigtableType.DEFAULT_COLUMN_FAMILY_NAME): T =
-      f.get(columnFamily, v, null)
+  def apply[T](_columnFamily: String, _timestampMicros: Long)(
+    implicit f: BigtableField[T]
+  ): BigtableType[T] = new BigtableType[T] {
+    override protected val columnFamily: String = _columnFamily
+    override protected val timestampMicros: Long = _timestampMicros
 
-    def to(v: T, columnFamily: String = BigtableType.DEFAULT_COLUMN_FAMILY_NAME,
-                 timestamp: Long = 0L,
-                 columnQualifier: String = null): Iterable[Mutation] =
-      f.put(columnQualifier, v).map(Mutations.newSetCellMutation(columnFamily, timestamp))
+    override def from(v: Row): T =
+      f.get(columnFamily, v, null)
 
+    override def to(v: T): Seq[Mutation] =
+      f.put(null, v).map { b =>
+        Mutation
+          .newBuilder()
+          .setSetCell(
+            b.setFamilyName(columnFamily).setTimestampMicros(timestampMicros)
+          )
+          .build()
+      }
   }
+
+  def mutationsToRow(key: ByteString, mutations: Seq[Mutation]): Row =
+    Row.create(
+      key,
+      mutations
+        .sortBy(_.getSetCell.getColumnQualifier.toStringUtf8)
+        .map(_.getSetCell)
+        .map { setCell =>
+          RowCell.create(
+            setCell.getFamilyName,
+            setCell.getColumnQualifier,
+            setCell.getTimestampMicros,
+            List.empty.asJava,
+            setCell.getValue
+          )
+        }
+        .asJava
+    )
+
+  def rowToMutations(row: Row): Seq[Mutation] =
+    row.getCells.asScala.map { cell =>
+      Mutation.newBuilder()
+        .setSetCell(SetCell.newBuilder()
+            .setFamilyName(cell.getFamily)
+            .setColumnQualifier(cell.getQualifier)
+            .setTimestampMicros(cell.getTimestamp)
+            .setValue(cell.getValue)
+        )
+        .build()
+    }
 }
 
 sealed trait BigtableField[T] extends Serializable {
   def get(columnFamily: String, v: Row, k: String): T
-  def put(k: String, v: T): Iterable[SetCell.Builder]
+  def put(k: String, v: T): Seq[SetCell.Builder]
 }
 
 object BigtableField {
   trait Primitive[T] extends BigtableField[T] {
-    type ValueT
```

Comment:
This `ValueT` is never used anywhere.

Reply:
oh yeah, this is a leftover form ExampleType